### PR TITLE
Fix problem with ephemeralRunner Succeeded state before build executed

### LINKED ISSUE #3527: Ephemeral runner gets stuck in Successful state
### Checks

- [X] I've already read https://docs.github.com/en/actions/hosting-your-own-runners/managing-self-hosted-runners-with-actions-runner-controller/troubleshooting-actions-runner-controller-errors and I'm sure my issue is not covered in the troubleshooting guide.
- [X] I am using charts that are officially provided

### Controller Version

0.9.0

### Deployment Method

Helm

### Checks

- [X] This isn't a question or user support case (For Q&A and community support, go to [Discussions](https://github.com/actions/actions-runner-controller/discussions)).
- [X] I've read the [Changelog](https://github.com/actions/actions-runner-controller/blob/master/docs/gha-runner-scale-set-controller/README.md#changelog) before submitting this issue and I'm sure it's not due to any recently-introduced backward-incompatible changes

### To Reproduce

```markdown
1. I have ARC installed in GKE in one namespace: controller and runnersets in the same one
2. All works as expected in most of the cases
3. From time to time (randomly) we observe such behavior:
- ephemeral runnerset gets patch , e.g. from 0 desired to 1 replica
- ephemeral runner is created, but immediately its status is changed to Succeeded and nothing happens - the workload is "stuck" on waiting for runner
```

### Describe the bug

In controller logs I see that it already "Found the runner with the same name" - it looks like the controller is performing reconcile twice for the same ephemeralrunner in almost the same time, the second run "removes" runner and makes it hung.

The runner is eventually not created, and the ephemeral runner gets to stage Succeeded and stuck until workflow is cancelled. 

We started to observe this behavior when we moved to faster cluster. 

### Describe the expected behavior

The controller should create runner always on ephemeral runner creation. 

### Additional Context

```yaml
N/A
```

### Controller Logs

```shell
https://gist.github.com/katarzynainit/ceccccde10d5454aa104d0f5a98f9b0d
```

### Runner Pod Logs

```shell
N/A
```

### PR DESCRIPTION
**Pull request by kr-sabre**
Fixes #3527

Seems that changing order of actions fixes the issues:
- first call GH services and update ephemeral runner status
- apply finalizers on CR

### CODE REVIEW

**Comment by kr-sabre** [comment]
But I'm wondering if returning result after each finalizer apply isn't more correct approach...
Please advise

File: controllers/actions.github.com/ephemeralrunner_controller.go
Comment (nikola-jokic) [inline controllers/actions.github.com/ephemeralrunner_controller.go]: Hey, I don't think that changing the registration and then patching the finalizers should be the right way to go. However, I noticed that the order in which we applied finalizers is incorrect.
We first need to apply the `ephemeralRunnerFinalizerName`, then the `ephemeralRunnerActionsFinalizerName`.

Would you be interested in trying out that approach?

Comment (kr-sabre) [reply controllers/actions.github.com/ephemeralrunner_controller.go]: Sure, will do the tests, but shouldn't we return Result also after ephemeralrunner.actions.github.com/runner-registration-finalizer apply?

Comment (nikola-jokic) [reply controllers/actions.github.com/ephemeralrunner_controller.go]: Good catch, we should!

Comment (kr-sabre) [reply controllers/actions.github.com/ephemeralrunner_controller.go]: Updated the code, please check now

I don't see such issues with the updated code, let me check a little longer - will be back with update on Monday

Comment (nikola-jokic) [reply controllers/actions.github.com/ephemeralrunner_controller.go]: Looks good to me. I'll wait for your feedback on Monday before merging it.

Comment (nikola-jokic) [reply controllers/actions.github.com/ephemeralrunner_controller.go]: Can you please also fix the test :relaxed: The order of finalizers is different now

Comment (kr-sabre) [reply controllers/actions.github.com/ephemeralrunner_controller.go]: > Looks good to me. I'll wait for your feedback on Monday before merging it.

We didn't observe the issue since Wednesday :)

**Review by nikola-jokic — APPROVED** [review]
LGTM! Thank you!

**Pull request merged**
